# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/gcc/sparcv789.patch
# Copyright (C) 2020 - 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- gcc-13-20230416/gcc/config.gcc.vanilla	2023-04-18 16:11:01.857535818 +0200
+++ gcc-13-20230416/gcc/config.gcc	2023-04-18 16:12:11.963537117 +0200
@@ -3351,7 +3351,7 @@
 	tm_file="${tm_file} elfos.h sparc/sysv4.h sparc/sp-elf.h sparc/rtemself.h rtems.h newlib-stdint.h"
 	tmake_file="${tmake_file} sparc/t-sparc sparc/t-rtems"
 	;;
-sparc-*-linux*)
+sparc-*-linux*|sparcv*-*-linux*)
 	tm_file="${tm_file} elfos.h sparc/sysv4.h gnu-user.h linux.h glibc-stdint.h sparc/tso.h"
 	extra_options="${extra_options} sparc/long-double-switch.opt"
 	case ${target} in
@@ -5057,7 +5057,7 @@
 		for which in cpu cpu_32 cpu_64 tune tune_32 tune_64; do
 			eval "val=\$with_$which"
 			case ${val} in
-			"" | sparc | sparcv9 | sparc64 \
+			"" | sparc | sparcv7 | sparcv8 | sparcv9 | sparc64 \
 			| v7 | cypress \
 			| v8 | supersparc | hypersparc | leon | leon3 | leon3v7 \
 			| sparclite | f930 | f934 | sparclite86x \
--- gcc-10.2.0/libgcc/config.host.vanilla	2020-10-03 14:28:28.829991995 +0200
+++ gcc-10.2.0/libgcc/config.host	2020-10-03 14:28:52.561991867 +0200
@@ -1363,7 +1363,7 @@
 	tmake_file="${tmake_file} t-fdpbit t-crtfm"
 	extra_parts="$extra_parts crti.o crtn.o crtfastmath.o"
 	;;
-sparc-*-linux*)		# SPARC's running GNU/Linux, libc6
+sparc-*-linux*|sparcv*-*-linux*)		# SPARC's running GNU/Linux, libc6
 	tmake_file="${tmake_file} t-crtfm"
 	if test "${host_address}" = 64; then
 		tmake_file="$tmake_file sparc/t-linux64"
